core/src/main/java/feign/querymap/FieldQueryMapEncoder.java

```diff
@@ -56,7 +61,7 @@ private ObjectParamMetadata getMetadata(Class<?> objectType) {
     ObjectParamMetadata metadata = classToMetadata.get(objectType);
     if (metadata == null) {
       metadata = ObjectParamMetadata.parseObjectType(objectType);
-      classToMetadata.put(objectType, metadata);
+      classToMetadata.putIfAbsent(objectType, metadata);
```

Comment:
Could this whole method be replaced with:
```
return classToMetadata.computeIfAbsent(objectType, ObjectParamMetadata::parseObjectType);
```
???

Reply:
I think it could,
I wasn’t sure on the refactor culture. Some like to make as few changes as possible.

happy to add this in

Reply:
Updated, 
the other change I thought was to the core of this loop to be the following, please let me know if its okay as I woulduse a pair:

```java
return metadata.objectFields.stream()
            .map(this::pairFieldAndOptionalValue)
            .filter(pair -> pair.right.isPresent())
            .collect(Collectors.toMap(pair -> createKeyFromField(pair.left), pair -> pair.right));
```

Its much smaller (personally I think it  reads better), however, I  understand  not to everyones  taste as some do not like the use of pair and also pairFieldAndOptionalValue is a hidden side effect where if the field.get fails will throw then EncodeException

Reply:
@rage-shadowman what are thoughts on this refactor?

Reply:
I think it all depends on how `field.get` fails.  The current loop is vulnerable to that now.  I don't have any issue with catching there and wrapping in an `EncodeException`, that fits with the `Encoder` interface, so it's not a hidden side-effect, so long as you add some comments around the check.

Reply:
If you're asking my opinion, I think that any time you simplify the code without breaking anything, you make it better. I think this PR looks more simplified.

As far as that stream usage goes, it looks more difficult to read at-a-glance to me, but the original isn't that easy to read at-a-glance either, and I tend to be biased against all but the simplest of streams usage.

As long as your stream doesn't modify anything from the input `metadata`, or from one of the earlier `map()`d records, then there aren't any side effects. Throwing a runtime exception due to an unexpected illegal state sounds reasonable.